pkg/kubectl/run.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubectl
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+	"github.com/golang/glog"
+)
+
+type BasicReplicationController struct{}
+
+func (BasicReplicationController) ParamNames() []GeneratorParam {
+	return []GeneratorParam{
+		{"labels", false},
+		{"name", true},
+		{"replicas", true},
+		{"image", true},
+		{"command", false},
+	}
+}
+
+func (BasicReplicationController) Generate(params map[string]string) (runtime.Object, error) {
+	labelString, found := params["labels"]
+	var labels map[string]string
+	if found && len(labelString) > 0 {
+		labels = ParseLabels(labelString)
+	} else {
+		labels = map[string]string{
+			"name": params["name"],
```

Comment:
How about
```labels = labels.SelectorFromSet(labels.Set{"name": name}.AsSelector())
```
instead of:
```
labels = map[string]string{
	     "name": params["name"]}
```